### ISSUE TITLE
Can't autocomplete list of strings when union string `| string` is present

### ISSUE THREAD

**Issue by usernamehw** [Microsoft/TypeScript#28743]
**TypeScript Version:**  3.2.1-dev.20181127

**Search Terms:**

Tried `Domain: Completion Lists` github label, didn't find a duplicate, there is probably one somewhere...

**Code**

```ts
type ListOrString = 'one' | 'two' | string;
const str: ListOrString = '';// Trigger autocomplete
```

**Expected behavior:**

Autocomplete shows
* one
* two

**Actual behavior:**

Only word suggestions.

**Comment by jack-williams** [comment]
Duplicate of #12687

The union type is simplified into a normal-form via subtype reduction; hovering over `ListOrString` will only show the supertype `string`.

**Comment by usernamehw** [comment]
So... 
* it was working at some point 
* it was reported and accepted to be a bug
* it was given an explanation `new is always better`
> The type "foo" | "bar" | string is equivalent to the string type and now gets simplified early on. So all we see when querying the type checker for the type of x is string. So, this can't be fixed.
* the issue got closed and locked so nobody could find it

**Comment by weswigham** [comment]
I've tagged the old issue with a domain label for future use, but it predates when we used domain labels, which is why we didn't have one on it. And most old issues without activity are closed/locked.

**Comment by usernamehw** [comment]
Can this be implemented as a TS plugin then? 

* Type checking should stay as it is 
* Editor autocomplete should have these items

**Comment by weswigham** [comment]
We don't really have plugins - sans editor-wide ls plugins that wrap the entire ls. They can do anything, since they just wrap and intercept the LS, it's just not always easy.
